pkg/kubefedctl/join.go

```diff
@@ -184,6 +184,17 @@ func (j *joinFederation) Run(cmdOut io.Writer, config util.FedConfig) error {
 		return err
 	}
 
+	// Set HostClusterContext as current context in kube config if it is not set
+	if j.HostClusterContext == "" {
+		clientConfig := config.GetClientConfig(j.HostClusterContext, j.Kubeconfig)
```

Comment:
The call on this line is identical to the one on line 168 which suggests calling `hostConfig.RawConfig()`.

Reply:
~~The call on this line is functionally identical to the one on line 168 which suggests calling `hostConfig.RawConfig()`.~~

Edit: I missed that `HostConfig` calls `GetClientConfig` internally and they return different types. Please ensure that `GetClientConfig` is only called once e.g.

```golang
hostClientConfig := ...
hostConfig, err := hostClientConfig.ClientConfig()
...
if j.HostClusterContext ... {
  rawConfig, err := ...
}
```

Reply:
done, thanks